test/e2e/infraprovider/providers/kind/kind.go

```diff
@@ -359,6 +361,85 @@ func (c *contextKind) getAttachedNetworks() (api.Networks, error) {
 	return attachedNetworks, nil
 }
 
+func (c *contextKind) SetupUnderlay(f *framework.Framework, underlay api.Underlay) error {
+	if underlay.LogicalNetworkName == "" {
+		return fmt.Errorf("underlay logical network name must be set")
+	}
+
+	if underlay.PhysicalNetworkName == "" {
+		underlay.PhysicalNetworkName = "underlay"
+	}
+
+	if underlay.BridgeName == "" {
+		underlay.BridgeName = secondaryBridge
+	}
+
+	const (
+		ovsNodeLabel = "app=ovnkube-node"
+	)
+
+	ovsPodList, err := f.ClientSet.CoreV1().Pods(deploymentconfig.Get().OVNKubernetesNamespace()).List(
+		context.Background(),
+		metav1.ListOptions{LabelSelector: ovsNodeLabel},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to list OVS pods: %w", err)
+	}
+
+	if len(ovsPodList.Items) == 0 {
+		return fmt.Errorf("no OVS pods found with label %s", ovsNodeLabel)
+	}
+
+	for _, ovsPod := range ovsPodList.Items {
+		if underlay.BridgeName != deploymentconfig.Get().ExternalBridgeName() {
+			underlayInterface, err := getNetworkInterface(ovsPod.Spec.NodeName, underlay.PhysicalNetworkName)
```

Comment:
So here we're getting the physical interface name directly by looking up what's plugged in the docker container from the specified network, instead of passing "eth0" and "eth1" all throughout the tests. Did I get it right?